syntax = "proto3";

package rtype;

message PayloadHeader {
    uint32 body_size = 1;
}

enum Type {
    PLAYER = 0;
    BULLET = 1;
    ENEMY = 2;
    WALL = 3;
    BONUS = 4;
}

message Payload {
    uint32 idSprite = 1;
    uint32 posX = 2;
    uint32 posY = 3;
    uint32 width = 4;
    uint32 height = 5;
}


enum EventType {
    MOVEUP = 0;
    MOVEDOWN = 1;
    MOVELEFT = 2;
    MOVERIGHT = 3;
    SHOOT = 4;
    QUIT = 5;
}


message Event {
    EventType event = 1;
}